include/drivers/clock_control.h

```diff
@@ -150,6 +151,39 @@ static inline int clock_control_off(const struct device *dev,
 	return api->off(dev, sys);
 }
 
+/**
+ * @brief Configure a source clock
+ *
+ * This function is non-blocking and can be called from any context.
+ * On success, the selected clock is configured as the client's source clock.
+ *
+ * It is client's reposnsibility to ensure that subsequent calls to the API
+ * provides the right information to enable clock_control driver to perform
+ * the right action (such as using the right clock source on _get_rate call).
+ *
+ * @param dev Device structure whose driver controls the clock
+ * @param sys Opaque data representing the clock
+ * @return 0 on success, negative errno on failure.
+ */
+static inline int clock_control_select(const struct device *dev,
```

Comment:
@nordic-krch Would you mind commenting this approval ?
I've added it on top of current PR, but I plan to raise it separately once we're aligned (and then rebase current PR on top)